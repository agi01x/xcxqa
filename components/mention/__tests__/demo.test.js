import demoTest from '../../../tests/shared/demoTest';

jest.mock('draft-js/lib/generateRandomKey', () => () => '123');

demoTest('mention', { skip: process.env.LIB_DIR === 'dist' });
