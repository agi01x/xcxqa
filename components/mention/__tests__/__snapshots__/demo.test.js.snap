// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/mention/demo/async.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/avatar.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/basic.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span>
                    <span
                      data-offset-key="123-0-0"
                    >
                      <span
                        data-text="true"
                      >
                        @afc163
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/controllder-simple.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
>
  <div
    class="ant-mention-editor oneline"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span>
                    <span
                      data-offset-key="123-0-0"
                    >
                      <span
                        data-text="true"
                      >
                        @afc163
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/controlled.md correctly 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col-6 ant-form-item-label"
    >
      <label
        class=""
        for="control-mention"
        title="Top coders"
      >
        Top coders
      </label>
    </div>
    <div
      class="ant-col-16 ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control has-success"
      >
        <span
          class="ant-form-item-children"
        >
          <div
            class="ant-mention-wrapper"
          >
            <div
              class="ant-mention-editor oneline"
            >
              <div
                class="ant-mention-toolbar"
              />
              <div
                class="ant-mention-editor-wrapper"
              >
                <div
                  class="DraftEditor-root"
                >
                  <div
                    class="DraftEditor-editorContainer"
                  >
                    <div
                      aria-describedby="placeholder-123"
                      class="notranslate public-DraftEditor-content"
                      contenteditable="true"
                      role="textbox"
                      spellcheck="false"
                      style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
                    >
                      <div
                        data-contents="true"
                      >
                        <div
                          class=""
                          data-block="true"
                          data-editor="123"
                          data-offset-key="123-0-0"
                        >
                          <div
                            class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                            data-offset-key="123-0-0"
                          >
                            <span>
                              <span
                                data-offset-key="123-0-0"
                              >
                                <span
                                  data-text="true"
                                >
                                  @afc163
                                </span>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col-14 ant-col-offset-6 ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control"
      >
        <span
          class="ant-form-item-children"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Submit
            </span>
          </button>
             
          <button
            class="ant-btn"
            type="button"
          >
            <span>
              Reset
            </span>
          </button>
        </span>
      </div>
    </div>
  </div>
</form>
`;

exports[`renders ./components/mention/demo/custom-tag.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="public-DraftEditorPlaceholder-root"
        >
          <div
            class="public-DraftEditorPlaceholder-inner"
            id="placeholder-123"
            style="white-space:pre-wrap"
          >
            @someone
          </div>
        </div>
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/multilines.md correctly 1`] = `
<div
  class="ant-mention-wrapper multilines"
  style="width:100%;height:100px"
>
  <div
    class="ant-mention-editor"
    style="width:100%;height:100px"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%;height:100px"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/multiple-trigger.md correctly 1`] = `
<div
  class="ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="public-DraftEditorPlaceholder-root"
        >
          <div
            class="public-DraftEditorPlaceholder-inner"
            id="placeholder-123"
            style="white-space:pre-wrap"
          >
            input @ to mention people, # to mention tag
          </div>
        </div>
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/placement.md correctly 1`] = `
<div
  class="ant-mention-placement-top ant-mention-wrapper"
  style="width:100%"
>
  <div
    class="ant-mention-editor oneline"
    style="width:100%"
  >
    <div
      class="ant-mention-toolbar"
    />
    <div
      class="ant-mention-editor-wrapper"
      style="width:100%"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-123"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="123"
                data-offset-key="123-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="123-0-0"
                >
                  <span
                    data-offset-key="123-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/mention/demo/popupContainer.md correctly 1`] = `
<button
  class="ant-btn ant-btn-primary"
  type="button"
>
  <span>
    Click Me
  </span>
</button>
`;

exports[`renders ./components/mention/demo/readonly.md correctly 1`] = `
<div>
  <div
    style="margin-bottom:10px"
  >
    <div
      class="ant-mention-wrapper disabled"
      style="width:100%"
    >
      <div
        class="ant-mention-editor readonly oneline"
        style="width:100%"
      >
        <div
          class="ant-mention-toolbar"
        />
        <div
          class="ant-mention-editor-wrapper"
          style="width:100%"
        >
          <div
            class="DraftEditor-root"
          >
            <div
              class="public-DraftEditorPlaceholder-root"
            >
              <div
                class="public-DraftEditorPlaceholder-inner"
                id="placeholder-123"
                style="white-space:pre-wrap"
              >
                this is disabled Mention
              </div>
            </div>
            <div
              class="DraftEditor-editorContainer"
            >
              <div
                aria-describedby="placeholder-123"
                class="public-DraftEditor-content"
                contenteditable="false"
                spellcheck="false"
                style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
              >
                <div
                  data-contents="true"
                >
                  <div
                    class=""
                    data-block="true"
                    data-editor="123"
                    data-offset-key="123-0-0"
                  >
                    <div
                      class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                      data-offset-key="123-0-0"
                    >
                      <span
                        data-offset-key="123-0-0"
                      >
                        <br
                          data-text="true"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <input
            style="width:0;opacity:0;border:0;position:absolute;left:0;top:0"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-mention-wrapper readonly"
    style="width:100%"
  >
    <div
      class="ant-mention-editor readonly oneline"
      style="width:100%"
    >
      <div
        class="ant-mention-toolbar"
      />
      <div
        class="ant-mention-editor-wrapper"
        style="width:100%"
      >
        <div
          class="DraftEditor-root"
        >
          <div
            class="public-DraftEditorPlaceholder-root"
          >
            <div
              class="public-DraftEditorPlaceholder-inner"
              id="placeholder-123"
              style="white-space:pre-wrap"
            >
              this is readOnly Mention
            </div>
          </div>
          <div
            class="DraftEditor-editorContainer"
          >
            <div
              aria-describedby="placeholder-123"
              class="public-DraftEditor-content"
              contenteditable="false"
              spellcheck="false"
              style="outline:none;user-select:text;-webkit-user-select:text;white-space:pre-wrap;word-wrap:break-word"
            >
              <div
                data-contents="true"
              >
                <div
                  class=""
                  data-block="true"
                  data-editor="123"
                  data-offset-key="123-0-0"
                >
                  <div
                    class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                    data-offset-key="123-0-0"
                  >
                    <span
                      data-offset-key="123-0-0"
                    >
                      <br
                        data-text="true"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <input
          style="width:0;opacity:0;border:0;position:absolute;left:0;top:0"
        />
      </div>
    </div>
  </div>
</div>
`;
